tests/std/tests/P0645R10_text_formatting_parse_contexts/test.cpp

```diff
@@ -2,11 +2,106 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
 #include <assert.h>
+#include <concepts>
 #include <format>
 #include <string_view>
 
-// TODO: fill in tests
+using namespace std;
+using namespace literals;
+
+template <class CharType>
+constexpr auto get_input() {
+    if constexpr (same_as<CharType, char>) {
+        return "First {} and second {} and third {}"sv;
+    } else {
+        return L"First {} and second {} and third {}"sv;
+    }
+}
+
+template <class CharType>
+constexpr bool ensure_is_constant_expression(const bool should_be_constant_expression) {
+    basic_format_parse_context<CharType> context{get_input<CharType>(), 1};
+    if (should_be_constant_expression) {
+        context.check_arg_id(0);
+    } else {
+        context.check_arg_id(1);
+    }
+    if (is_constant_evaluated()) {
+        return true;
+    } else {
+        return false;
+    }

```

Comment:
Instead of `if (X) { return true; } else { return false; }`, saying `return X;` is simpler.

Reply:
Yeah, I was unsure about tue `is_constant _evaluated` call.